include/BCFKeyValueData.h

```diff
@@ -50,19 +50,22 @@ class BCFKeyValueData : public Metadata, public BCFData {
     Status dataset_header(const std::string& dataset,
                               std::shared_ptr<const bcf_hdr_t>& hdr) const override;
     Status dataset_range(const std::string& dataset, const bcf_hdr_t* hdr,
-                         const range& pos, unsigned min_alleles,
+                         const range& pos,
+                         bool predicate(const bcf_hdr_t*, bcf1_t*),
```

Comment:
Elsewhere you use the `bcf_predicate` typedef but here you write it out. Can we do one or the other consistently? The type might be short enough that it wouldn't be awful to write it everywhere explicitly, but the choice is up to you.